test/lein_monolith/task/with_dependency_set_test.clj

```diff
@@ -0,0 +1,72 @@
+(ns lein-monolith.task.with-dependency-set-test
+  (:require
+    [clojure.test :refer [deftest testing is]]
+    [lein-monolith.plugin :as plugin]
+    [lein-monolith.task.with-dependency-set :as wds]
+    [lein-monolith.test-utils :refer [use-example-project]]
+    [leiningen.core.main :as lein]
+    [leiningen.core.project :as project]))
+
+
+(use-example-project)
+
+
+(defn- remove-unstable
+  [project]
+  (assoc-in project [:monolith :project-selectors :unstable] "non-deterministic"))
+
+
+(defn- read-project
+  [path]
+  (-> (project/read-raw path)
+      (plugin/add-middleware)
+      (project/init-project)))
```

Comment:
I'm using this fn rather than `read-example-project` because there's a weird circular behavior when `lein-monolith` isn't in your user profiles. This forcibly loads the project as a raw, unevaluated project, forcibly adds the monolith middleware, and then re-inits the project
[Slack thread for more context](https://amperity.slack.com/archives/C01MAB8KMTP/p1716499522371359?thread_ts=1716493862.855569&cid=C01MAB8KMTP)